frontend/projects/upgrade/src/app/features/dashboard/feature-flags/modals/upsert-feature-flag-modal/upsert-feature-flag-modal.component.ts

```diff
@@ -0,0 +1,242 @@
+  selector: 'upsert-add-feature-flag-modal',
+  standalone: true,
+  imports: [
+    CommonModalComponent,
+    MatCardModule,
+    MatFormFieldModule,
+    MatInputModule,
+    MatButtonModule,
+    MatDialogTitle,
+    MatDialogContent,
+    MatDialogActions,
+    MatDialogClose,
+    MatSelectModule,
+    CommonModule,
+    NgTemplateOutlet,
+    MatIcon,
+    ReactiveFormsModule,
+    TranslateModule,
+    CommonTagsInputComponent,
+  ],
+  templateUrl: './upsert-feature-flag-modal.component.html',
+  styleUrl: './upsert-feature-flag-modal.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class UpsertFeatureFlagModalComponent {
+  isLoadingUpsertFeatureFlag$ = this.featureFlagsService.isLoadingUpsertFeatureFlag$;
+  isSelectedFeatureFlagUpdated$ = this.featureFlagsService.isSelectedFeatureFlagUpdated$;
+  selectedFlag$ = this.featureFlagsService.selectedFeatureFlag$;
+  appContexts$ = this.featureFlagsService.appContexts$;
+
+  subscriptions = new Subscription();
+  isInitialFormValueChanged$: Observable<boolean>;
+  isPrimaryButtonDisabled$: Observable<boolean>;
+
+  initialFormValues$ = new BehaviorSubject<FeatureFlagFormData>(null);
+
+  featureFlagForm: FormGroup;
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA)
+    public config: CommonModalConfig<UpsertFeatureFlagParams>,
+    public dialog: MatDialog,
+    private formBuilder: FormBuilder,
+    private featureFlagsService: FeatureFlagsService,
+    private experimentService: ExperimentService,
+    private formHelpersService: CommonFormHelpersService,
+    public dialogRef: MatDialogRef<UpsertFeatureFlagModalComponent>
+  ) {}
+
+  ngOnInit(): void {
+    this.experimentService.fetchContextMetaData();
+    this.createFeatureFlagForm();
+    this.listenForFeatureFlagGetUpdated();
+    this.listenOnNameChangesToUpdateKey();
+    this.listenForIsInitialFormValueChanged();
+    this.listenForPrimaryButtonDisabled();
+  }
+
+  createFeatureFlagForm(): void {
+    const { sourceFlag, action } = this.config.params;
+    const initialValues = this.deriveInitialFormValues(sourceFlag, action);
+
+    this.featureFlagForm = this.formBuilder.group({
+      name: [initialValues.name, Validators.required],
+      key: [initialValues.key, Validators.required],
+      description: [initialValues.description],
+      appContext: [initialValues.appContext, Validators.required],
+      tags: [initialValues.tags],
+    });
+
+    this.initialFormValues$.next(this.featureFlagForm.value);
+  }
+
+  deriveInitialFormValues(sourceFlag: FeatureFlag, action: string) {
+    const name = this.deriveName(sourceFlag, action);
+    const key = this.deriveKey(sourceFlag, action);
+    const description = this.deriveDescription(sourceFlag);
+    const appContext = this.deriveAppContext(sourceFlag);
+    const tags = this.deriveTags(sourceFlag);
+
+    return { name, key, description, appContext, tags };
+  }
+
+  deriveName(sourceFlag: FeatureFlag, action: string): string {
```

Comment:
why derive values separately for every property?

Reply:
what's your suggested code improvement?

Reply:
can u try this?
`deriveInitialFormValues(sourceFlag: FeatureFlag, action: string) {
  return {
    name: action === UPSERT_FEATURE_FLAG_ACTION.EDIT ? sourceFlag?.name : '',
    key: action === UPSERT_FEATURE_FLAG_ACTION.EDIT ? sourceFlag?.key : '',
    description: sourceFlag?.description || '',
    appContext: sourceFlag?.context?.[0] || '',
    tags: sourceFlag?.tags || [],
  };
}`

Reply:
That's exactly what I'd like to avoid doing, what's the benefit in jamming all that in there?

Reply:
I'll ask in the dev mtg, I'm curious what others say, I guess I have a very different coding style than the rest of the team in general, I want to do what is comfortable for everyone.